src/app/shared/components/workshop-card/workshop-card.component.ts

```diff
@@ -108,6 +113,40 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
     this.isFavorite = !this.isFavorite;
   }
 
+  onChangeWorkshopStatus(status: string, type: ModalConfirmationType): void {
+    const dialogRef = this.dialog.open(ConfirmationModalWindowComponent, {
+      width: Constants.MODAL_SMALL,
+      data: {
+        type: type,
+      },
+    });
+
+    dialogRef.afterClosed().pipe(
+      switchMap((res) =>
+        res
+          ? this.store
+              .dispatch(
+                new UpdateStatus({
+                  workshopId: this.workshopData.workshopId,
+                  status,
+                })
+              )
+              .pipe(
+                res
+                  ? switchMap((res) =>
+                      this.store.dispatch(
+                        new GetWorkshopsByProviderId(
+                          this.workshopData.providerId
```

Comment:
you can do that is the state (the action should be successful, so it has to be done on UpdateWorkshopStatusSuccess